Mathlib/LinearAlgebra/Projection.lean

```diff
@@ -444,3 +480,68 @@ theorem IsProj.eq_conj_prodMap {f : E →ₗ[R] E} (h : IsProj p f) :
 end LinearMap
 
 end CommRing
+
+section
```

Comment:
I think these results should be in the `LinearMap` namespace, even if that prevents the use of dot-notation. Otherwise a theorem name like `IsIdempotentElem.mem_range_iff` gives no context that the theorem is about linear maps as opposed to some other kind of map (and this could potentially lead to a name clash in the future).